src/modules/gateways/paymentwall.php

```diff
@@ -25,37 +28,111 @@ function init_paymentwall_config($params)
-                Paymentwall_Product::TYPE_FIXED
-            )
-        ),
-        array_merge(
+                $params["description"] . ' + Setup Fee',
+                Paymentwall_Product::TYPE_SUBSCRIPTION,
+                1,
+                $periodType,
+                true
+            );
+            $hasTrial = true;
+        }
+
+        $subscriptionProduct = new Paymentwall_Product(
+            $primaryServiceId,
+            $recurring['recurringamount'],
+            $params['currency'],
+            $params["description"],
+            Paymentwall_Product::TYPE_SUBSCRIPTION,
+            $recurring['recurringcycleperiod'],
+            $periodType,
+            true,
+            $trialProduct
+        );
+    }
+
+    $onetimeProduct = new Paymentwall_Product(
+        (int)$params['invoiceid'],
+        $params['amount'],
+        $params['currency'],
+        $params["description"],
+        Paymentwall_Product::TYPE_FIXED
+    );
+
+    init_paymentwall_config($params);
+
+    if ($subscriptionProduct && (!$params['forceOneTime'] || $params['forceSubscription'])) {
+        $subscriptionWidget = new Paymentwall_Widget(
+            $params['clientdetails']['email'],
+            $params['widget'],
             array(
-                'integration_module' => 'whmcs',
-                'is_test' => (isset($params['isTest']) && $params['isTest'] != '') ? 1 : 0
+                $subscriptionProduct
             ),
-            get_user_profile_data($params)
-        )
-    );
-    $widgetUrl = $params['systemurl'] . '/paymentwallwidget.php';
-    $code = '<form method=POST action="' . $widgetUrl . '">
-        <input type="hidden" name="data" value="' . encrypt($widget->getHtmlCode(array(
-            'width' => '100%',
-            'height' => '500'
-        ))) . '" />
-        <button type="submit">Pay via <img src="' . $params['systemurl'] . '/images/paymentwall/paymentwall_logo.png" alt="Pay via Paymentwall" /></button>
+            array_merge(
+                array(
+                    'integration_module' => 'whmcs',
+                    'is_test' => (isset($params['isTest']) && $params['isTest'] != '') ? 1 : 0,
+                    'hide_post_trial_good' => $hasTrial ? 1 : 0,
+                ),
+                get_user_profile_data($params)
+            )
+        );
+
+        $code .= '<form method=POST action="' . $params['systemurl'] . '/paymentwallwidget.php" style="display:inline;">
+        <input type="hidden" name="data" value="' . encrypt($subscriptionWidget->getHtmlCode(array(
+                'width' => '100%',
+                'height' => '500'
+            ))) . '" />
+            <input type="image" src="' . $params['systemurl'] . '/images/paymentwall/subscribe.png" onClick="this.form.submit()"/>
     </form>';
+    }
 
+    if ((!$params['forceSubscription'] && $recurring) || !$recurring) {
+        $onetimeWidget = new Paymentwall_Widget(
+            $params['clientdetails']['email'],

```

Comment:
can we send user id? instead email

also better to send email in `email` parameter, not in `uid`

https://www.paymentwall.com/en/documentation/Digital-Goods-API/710#paymentwall_widget_call_flexible_widget_call